zerver/lib/actions.py

```diff
@@ -766,6 +766,32 @@ def do_scrub_realm(realm: Realm) -> None:
     RealmAuditLog.objects.create(realm=realm, event_time=timezone_now(),
                                  event_type=RealmAuditLog.REALM_SCRUBBED)
 
+def do_delete_user(user_profile: UserProfile) -> None:
+    do_deactivate_user(user_profile)
+    if user_profile.realm.is_zephyr_mirror_realm:  # nocoverage
+        # TODO: What should actually be done in this case?
+        return
+
+    subscribed_huddle_recipient_ids = set(
+        Subscription.objects.filter(user_profile=user_profile, recipient__type=Recipient.HUDDLE)
+        .values_list('recipient_id', flat=True)
+    )
+    user_id = user_profile.id
+    realm = user_profile.realm
+    personal_recipient = user_profile.recipient
+
+    user_profile.delete()
```

Comment:
Let's add a comment explaining why `personal_recipient` is one of the few things we need to delete explicitly, namely, that the deletion doesn't cascade.  And maybe also that `personal_recipient` will cascade to deleting all PMs that had been sent to the user.

(Is that our intent?  Or should we be instead just reattaching the `personal_recipient` to the new fake user?).

Reply:
Added a comment. I think it makes sense to delete PMs to the user if we're deleting messages from the user - in the spirit of cleaning up the obvious personal data, which PMs to the user probably should be considered.